wolfcrypt/src/port/psa/psa_pkcbs.c

```diff
@@ -0,0 +1,546 @@
+/* psa_pkcbs.c
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#if defined(WOLFSSL_HAVE_PSA)
+#if defined(HAVE_PK_CALLBACKS)
+#if !defined(WOLFSSL_PSA_NO_PKCBS)
+
+#include <psa/crypto.h>
+
+#include <wolfssl/wolfcrypt/port/psa/psa.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/wolfcrypt/asn.h>
+#include <wolfssl/wolfcrypt/logging.h>
+
+#ifdef HAVE_ECC
+
+#include <wolfssl/wolfcrypt/ecc.h>
+
+static int psa_ecc_get_curve_info_from_curve_id(int ecc_curve,
+                                                psa_ecc_family_t *family,
+                                                size_t *key_size)
+{
+    if (ecc_curve != ECC_SECP256R1)
```

Comment:
Note: If only SECP256R1 is supported then the `MAX_ECC_BYTES` is 32. Curious why not adding support for additional curves? What happens for the non-SECP256R1 curves? Instead of `WOLFSSL_NOT_IMPLEMENTED` consider `NOT_COMPILED_IN`. No action required here, just curious observation.

Reply:
The idea was to have a first PR with the minimal set of features to have a TLS 1.3 ciphersuite working and then to add other curves, hash algs and ciphers in smaller PRs. I thought this make the review process easier.